datafusion-federation/src/sql/mod.rs

```diff
@@ -0,0 +1,597 @@
+mod analyzer;
+pub mod ast_analyzer;
+mod executor;
+mod schema;
+mod table;
+mod table_reference;
+
+use std::{any::Any, fmt, sync::Arc, vec};
+
+use analyzer::{collect_known_rewrites, RewriteTableScanAnalyzer};
+use ast_analyzer::{AstAnalyzer, RewriteMultiTableReference};
+use async_trait::async_trait;
+use datafusion::{
+    arrow::datatypes::{Schema, SchemaRef},
+    common::tree_node::TreeNode,
+    config::ConfigOptions,
+    error::{DataFusionError, Result},
+    execution::{context::SessionState, TaskContext},
+    logical_expr::{Extension, LogicalPlan},
+    optimizer::{Analyzer, AnalyzerRule},
+    physical_expr::EquivalenceProperties,
+    physical_plan::{
+        execution_plan::{Boundedness, EmissionType},
+        DisplayAs, DisplayFormatType, ExecutionPlan, Partitioning, PlanProperties,
+        SendableRecordBatchStream,
+    },
+    sql::{sqlparser::ast::Statement, unparser::Unparser},
+};
+
+pub use executor::{LogicalOptimizer, SQLExecutor, SQLExecutorRef};
+pub use schema::{MultiSchemaProvider, SQLSchemaProvider};
+pub use table::{RemoteTable, SQLTableSource};
+pub use table_reference::RemoteTableRef;
+
+use crate::{
+    get_table_source, schema_cast, FederatedPlanNode, FederationPlanner, FederationProvider,
+};
+
+// SQLFederationProvider provides federation to SQL DMBSs.
+#[derive(Debug)]
+pub struct SQLFederationProvider {
```

Comment:
Upstream removed the `datafusion-federation-sql` crate and just made it a feature in the main crate. This is the move of that logic